spring-cloud-aws-autoconfigure/src/test/java/io/awspring/cloud/autoconfigure/sqs/it/SqsAutoConfigurationIntegrationTest.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.autoconfigure.sqs.it;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testcontainers.containers.localstack.LocalStackContainer.Service.SQS;
+
+import io.awspring.cloud.autoconfigure.core.AwsAutoConfiguration;
+import io.awspring.cloud.autoconfigure.core.CredentialsProviderAutoConfiguration;
+import io.awspring.cloud.autoconfigure.core.RegionProviderAutoConfiguration;
+import io.awspring.cloud.autoconfigure.sqs.SqsAutoConfiguration;
+import io.awspring.cloud.sqs.annotation.SqsListener;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.util.TestPropertyValues;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.services.sqs.SqsAsyncClient;
+
+/**
+ * Integration tests for SQS.
+ *
+ * @author Tomaz Fernandes
+ * @since 3.0
+ */
+@SpringBootTest
+@Testcontainers
+class SqsAutoConfigurationIntegrationTest {
+
+	private static final String REGION = "eu-west-1";
+
+	private static final String QUEUE_NAME = "my_queue_name";
+
+	private static final String PAYLOAD = "Test";
+
+	private AnnotationConfigApplicationContext context;
+
+	@Container
+	static LocalStackContainer localstack = new LocalStackContainer(
+			DockerImageName.parse("localstack/localstack:0.14.2")).withServices(SQS).withReuse(true);
```

Comment:
```suggestion
			DockerImageName.parse("localstack/localstack:0.14.2")).withServices(SQS);
```

unless we have `testcontainers.reuse.enable=true` at `~/.testcontainers.properties` then this make sense. What reuse does? It will keep one container running and reuse but the container will not stop. In our case, I don't think is the case.

Reply:
If I understand correctly, the `reusable=true` makes running tests locally faster, and should not be a problem for the CI. Is this correct? That being said, I'm ok with not setting it if that's better.

As for the `DockerImageName`, it seems passing the String directly to the constructor has been deprecated.

```java
    @Deprecated
    public LocalStackContainer(String version) {
        this(DEFAULT_IMAGE_NAME.withTag(version));
    }
```

Am I missing something?

Reply:
right about DockerImageName. Forgot it is different in LocalStack for what it is usually offered.

About reuse containers, there is no official documentation yet and btw it is an experimental feature. You can read this [blogpost](https://rieckpil.de/reuse-containers-with-testcontainers-for-fast-integration-tests/) which explains reuse containers feature very well.

Reply:
Oh, ok, now I see it, it might leave the CI container running.  I'll remove it, thanks.

Reply:
Or, actually, it won't leave the container running in the CI since we don't have the `testcontainers.reuse.enable=true ` property set, but I wasn't really aware this was leaving a container running in my machine 😄 

I thought this was more about not creating new container instances for each test.

Thanks for the heads up.